templates/typescript_gapic/src/$version/$service_client.ts.njk

```diff
@@ -284,11 +284,23 @@ export class {{ service.name }}Client {
     // This API contains "long-running operations", which return a
     // an Operation object that allows for tracking of the operation,
     // rather than holding a request open.
-
-    this.operationsClient = this._gaxModule.lro({
+    const lroOptions = {
       auth: this.auth,
       grpc: 'grpc' in this._gaxGrpc ? this._gaxGrpc.grpc : undefined
-    }).operationsClient(opts);
+    };
+{%- if not api.legacyProtoLoad %}
+    if (opts.fallback === 'rest') {
+      Object.assign(lroOptions, {
+{%- if (service.longRunning.length > 0) or (service.bundleConfigs.length > 0) %}
+        protoJson: protoFilesRoot,
+{%- else %}
+        protoJson: this._gaxModule.protobuf.Root.fromJSON(jsonProtos),
```

Comment:
Can you clarify why this is needed? If I'm reading the code above correctly, `protoFilesRoot` is assigned to this exact value on line 280. Why do we need this if-else condition here?

Reply:
There is one type of scenario that `service.longrunning.length === 0`, then `protoFilesRoot` variable is never defined. See line [#271](https://github.com/googleapis/gapic-generator-typescript/blob/main/templates/typescript_gapic/src/%24version/%24service_client.ts.njk#L271)

Reply:
[showcase/identify_client](https://github.com/googleapis/gapic-generator-typescript/blob/main/baselines/showcase/src/v1beta1/identity_client.ts.baseline)
and [showcase/test_client](https://github.com/googleapis/gapic-generator-typescript/blob/main/baselines/showcase/src/v1beta1/testing_client.ts.baseline) has no `protoFileRoot` variable.

Reply:
Oh, I see now. Well, from what I see, now we need `protoFilesRoot` in any case (other than the legacy proto load for Ads), so we can as well define it in line 271 in any case, regardless of if it's LRO or not. This will make the code cleaner.

Reply:
Cool. I add the `protoFilesRoot` in that case.